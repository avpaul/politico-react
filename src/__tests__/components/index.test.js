import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import localStorageMock from '../__mocks__/localStorage';
import Badge from '../../components/badge/badge';
import Navbar from '../../components/navbarComponent/navbar';
import ResultCard from '../../components/resultsComponent/resultCard';
import Home from '../../components/homeComponent/home';
import Login from '../../components/loginComponent/login';
import Signup from '../../components/signupComponent/signup';

const mockStore = configureStore();

Object.defineProperty(global, 'localStorage', { value: localStorageMock });

describe('components', () => {
  it('should render a badge component', () => {
    const message = { text: 'Profile updated', type: 'success' };
    const wrapper = shallow(<Badge message={message.text} type={message.type} duration={6000} />);
    expect(wrapper.find('span').text()).toEqual(message.text);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a navbar component', () => {
    const wrapper = shallow(<Navbar />);
    expect(
      wrapper
        .find('img')
        .first()
        .hasClass('logo-img'),
    ).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the home component', () => {
    const wrapper = shallow(<Home />);
    wrapper.setState({ auth: false });
    expect(
      wrapper
        .find('.btn.btn-signup')
        .dive()
        .text(),
    ).toEqual('');
  });

  it('render result card', () => {
    const wrapper = shallow(<ResultCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render result card', () => {
    const store = mockStore({ redirectTo: {}, errors: {} });
    const wrapper = shallow(
      <Provider store={store}>
        <Signup />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render result card', () => {
    const store = mockStore({ redirectTo: {}, errors: {} });
    const wrapper = shallow(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
