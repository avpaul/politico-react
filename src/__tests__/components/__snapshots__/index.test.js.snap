// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components render result card 1`] = `ShallowWrapper {}`;

exports[`components render result card 2`] = `ShallowWrapper {}`;

exports[`components render result card 3`] = `ShallowWrapper {}`;

exports[`components should render a badge component 1`] = `ShallowWrapper {}`;

exports[`components should render a navbar component 1`] = `ShallowWrapper {}`;
